report-generator/src/main/java/com/teamscale/report/testwise/jacoco/cache/ProbesCache.java

```diff
@@ -59,14 +60,16 @@ public boolean containsClassId(long classId) {
 	 * Converts the given {@link ExecutionData} to {@link FileCoverageBuilder} using the cached lookups or null if the class
 	 * file of this class has not been included in the analysis or was not covered.
 	 */
-	public FileCoverageBuilder getCoverage(ExecutionData executionData) throws CoverageGenerationException {
+	public FileCoverageBuilder getCoverage(ExecutionData executionData, Predicate<String> locationIncludeFilter) throws CoverageGenerationException {
 		long classId = executionData.getId();
 		if (!containsClassId(classId)) {
-			logger.debug(
-					"Found coverage for a class " + executionData
-							.getName() + " that was not provided. Either you did not provide " +
-							"all relevant class files or you did not adjust the include/exclude filters on the agent to exclude " +
-							"coverage from irrelevant code.");
+			if (locationIncludeFilter.test(executionData.getName())) {
```

Comment:
I take that if `!containsClassId() && !locationIncludeFilter.test()` then the filter and the cache are in sync and everything is alright, therefore we don't need to log anything, right?

Reply:
Exactly